File: src/main/java/io/github/kingg22/api/vacunas/panama/service/ITokenService.java
package io.github.kingg22.api.vacunas.panama.service;

import io.github.kingg22.api.vacunas.panama.web.dto.UsuarioDto;
import jakarta.validation.constraints.NotNull;
import java.io.Serializable;
import java.util.Map;

public interface ITokenService {
    Map<String, Serializable> generateTokens(
            @org.jetbrains.annotations.NotNull @NotNull UsuarioDto usuarioDto,
            Map<String, Serializable> idsAdicionales);

    boolean isAccessTokenValid(
            @org.jetbrains.annotations.NotNull @NotNull String userId,
            @org.jetbrains.annotations.NotNull @NotNull String tokenId);

    boolean isRefreshTokenValid(
            @org.jetbrains.annotations.NotNull @NotNull String userId,
            @org.jetbrains.annotations.NotNull @NotNull String tokenId);
}

File: src/main/kotlin/io/github/kingg22/api/vacunas/panama/service/ITokenService.kt
package io.github.kingg22.api.vacunas.panama.service

import io.github.kingg22.api.vacunas.panama.web.dto.UsuarioDto
import jakarta.validation.constraints.NotNull
import java.io.Serializable

interface ITokenService {
    fun generateTokens(
        @NotNull usuarioDto: UsuarioDto,
        idsAdicionales: Map<String, Serializable>,
    ): Map<String, Serializable>

    fun isAccessTokenValid(@NotNull userId: String, @NotNull tokenId: String): Boolean

    fun isRefreshTokenValid(@NotNull userId: String, @NotNull tokenId: String): Boolean
}
